File: src/weaponinventorymod/core/StockSortMode.java
package weaponinventorymod.core;

public enum StockSortMode {
    NEED("Stock"),
    NAME("Name"),
    PRICE("Price");

    private final String label;

    StockSortMode(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }

    public StockSortMode next() {
        int nextOrdinal = ordinal() + 1;
        StockSortMode[] values = values();
        return values[nextOrdinal >= values.length ? 0 : nextOrdinal];
    }

    public static StockSortMode fromConfig(String value) {
        if (value == null) {
            return NEED;
        }
        String normalized = value.trim().toUpperCase().replace('-', '_').replace(' ', '_');
        if ("PURCHASABLE".equals(normalized) || "FOR_SALE".equals(normalized) || "OWNED".equals(normalized)) {
            return NEED;
        }
        if ("COST".equals(normalized)) {
            return PRICE;
        }
        for (StockSortMode mode : values()) {
            if (mode.name().equals(normalized)) {
                return mode;
            }
        }
        return NEED;
    }
}

File: src/main/kotlin/weaponsprocurement/core/StockSortMode.kt
package weaponsprocurement.core

enum class StockSortMode(val label: String) {
    NEED("Stock"),
    NAME("Name"),
    PRICE("Price");

    fun next(): StockSortMode {
        val values = values()
        val nextOrdinal = ordinal + 1
        return values[if (nextOrdinal >= values.size) 0 else nextOrdinal]
    }

    companion object {
        @JvmStatic
        fun fromConfig(value: String?): StockSortMode {
            if (value == null) return NEED
            val normalized = value.trim().uppercase().replace('-', '_').replace(' ', '_')
            if (normalized == "PURCHASABLE" || normalized == "FOR_SALE" || normalized == "OWNED") {
                return NEED
            }
            if (normalized == "COST") return PRICE
            for (mode in values()) {
                if (mode.name == normalized) return mode
            }
            return NEED
        }
    }
}
